### PR TITLE
Ensure latest version tag is merged into master branch

### LINKED ISSUE #67: git merge -s ours <latest tag>  upon finishing processing of the dandiset
Extract of https://github.com/dandi/dandisets/issues/58#issuecomment-898715301 -- I quickly changed my mind as I think for all means and purposes, and with cron job running frequently enough it would look not that bad and definitely would be useful for any `git describe` junkies as I am.

We should should be sure to not merge some older tag, which somehow appeared whenever a newer one was already merged.

So pretty much it should be 

merge the latest (according to semver) of all the tags (from `git tag`) if `git describe` on branch is either null or older in semver.

WDYT @jwodder -- any hidden stones?

### PR DESCRIPTION
**Pull request by jwodder**
Closes #67.

### CODE REVIEW

File: tools/backups2datalad.req.txt [hunk 1]
```diff
@@ -7,4 +7,5 @@ datalad
 #fscacher
 humanize
 morecontext ~= 0.1
+packaging
```
Comment (yarikoptic) [inline tools/backups2datalad.req.txt:10]: Why not to use semantic-version package we already use in dandi CLI?

Comment (jwodder) [reply tools/backups2datalad.req.txt:10]: We also use `packaging` in dandi-cli, and as long as version IDs are of the form `N.N.N`, the two packages will give the same results.

Comment (yarikoptic) [reply tools/backups2datalad.req.txt:10]: indeed we do, but IMHO for no good reason:
```
$> git grep packaging
dandi/cli/cmd_shell_completion.py:from packaging.version import Version
setup.cfg:    packaging

$> git grep semantic_version
dandi/pynwb_utils.py:import semantic_version
dandi/pynwb_utils.py:    if isinstance(v, str) and not semantic_version.validate(v):
dandi/tests/test_metadata.py:from semantic_version import Version
dandi/tests/test_utils.py:from semantic_version import Version
dandi/utils.py:from semantic_version import Version
```

do you remember why to start using `packaging.version.Version`? if there is no advantage/need, I would say we should just use `semantic_version` and also remove `packaging` as a requirement for dandi-cli

Comment (jwodder) [reply tools/backups2datalad.req.txt:10]: @yarikoptic dandi uses `packaging.version.Version` in order to test the version of Click installed.  Python package versions follow a different format than semver, and so `packaging` is the correct tool for the job there.

**Comment by yarikoptic** [comment]
Please add into the test testing for the correct merge to happen.

File: tools/backups2datalad.py [hunk 2]
```diff
@@ -446,13 +448,35 @@ def get_file_bucket_url(self, asset: RemoteAsset) -> str:
...
+                m = re.fullmatch(
+                    r"(?P<tag>.+)-(?P<distance>[0-9]+)-g(?P<rev>[0-9a-f]+)?",
+                    description,
+                )
+                assert m, f"Could not parse `git describe` output: {description!r}"
+                merge = Version(latest) > Version(m["tag"])
+            if merge:
+                log.debug("Running: git merge -s ours %s", shlex.quote(latest))
+                subprocess.run(
+                    ["git", "merge", "-s", "ours", latest], cwd=ds.path, check=True
```
Comment (yarikoptic) [inline tools/backups2datalad.py:476]: I have tried it merged into #68 on 000029 and got an editor suggesting to edit the merge commit message. Please disable interactions by providing a merge message to `merge` invocation .  Also make that merge message mention that it is "no change" merge, e.g.

```suggestion
                    ["git", "merge", "-s", "ours", "-m", f"Merge '{latest}' into the branch with drafts (no differences in content merged)", latest], cwd=ds.path, check=True
```

or alike.

**Review by yarikoptic — CHANGES_REQUESTED** [review]
provide custom msg + please add some rudimentary test

**Comment by jwodder** [comment]
@yarikoptic I've added a commit message for the merge command, but adding more tests will be easier if #68 is merged first.

**Comment by yarikoptic** [comment]
#68 is merged now, thank you!

**Comment by jwodder** [comment]
@yarikoptic Tests updated.

**Comment by yarikoptic** [comment]
Thank you! look good, will test merged "in deployment".

**Pull request merged**